genesis/parse.go

```diff
@@ -0,0 +1,153 @@
+package genesis
+
+import (
+	"encoding/json"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk"
+	"github.com/pkg/errors"
+
+	cmn "github.com/tendermint/tmlibs/common"
+)
+
+// Option just holds module/key/value triples from
+// parsing the genesis file
+type Option struct {
+	Module string
+	Key    string
+	Value  string
+}
+
+// InitStater is anything that can handle app options
+// from genesis file. Setting the merkle store, config options,
+// or anything else
+type InitStater interface {
+	InitState(module, key, value string) error
+}
+
+// Load parses the genesis file and sets the initial
+// state based on that
+func Load(app InitStater, filePath string) error {
+	opts, err := GetOptions(filePath)
+	if err != nil {
+		return err
+	}
+
+	// execute all the genesis init options
+	// abort on any error
+	for _, opt := range opts {
+		err = app.InitState(opt.Module, opt.Key, opt.Value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetOptions parses the genesis file in a format
+// that can easily be handed into InitStaters
+func GetOptions(path string) ([]Option, error) {
+	genDoc, err := load(path)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := genDoc.AppOptions
+	cnt := 1 + len(opts.Accounts) + len(opts.pluginOptions)
+	res := make([]Option, cnt)
```

Comment:
If you do `make([]Option, 0, cnt)` then you can use `append` and don't need to have an `i` variable, and it's the same speed.

Reply:
nice tip